### PR TITLE
Jetty 9.4.x jdbc session test docker

### PR DESCRIPTION
**Pull request by olamy**
reopen of #4905 with correct target branch (jetty-9.4.x)

### CODE REVIEW

**Comment by olamy** [comment]
reopen from #4905 with correct target branch (jetty-9.4.x)

File: tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/JdbcTestHelper.java [hunk 1]
```diff
@@ -60,6 +68,39 @@
...
+        catch (Exception e)
+        {
+            LOG.error(e.getMessage(), e);
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
     static
     {
         System.setProperty("derby.system.home", MavenTestingUtils.getTargetFile("test-derby").getAbsolutePath());
```
Comment (janbartel) [inline tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/JdbcTestHelper.java]: Still need to delete this line setting the derby property.

File: tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/SessionTableSchemaTest.java [hunk 2]
```diff
@@ -119,8 +117,10 @@ public void testLoad()
         _da.initialize();
         _tableSchema.prepareTables();
 
+        String id = Long.toString(System.nanoTime());
```
Comment (janbartel) [inline tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/SessionTableSchemaTest.java:122]: Just out of interest, if the tables are dropped after every test, you don't need to make the id unique. Fine either way.

File: tests/test-sessions/test-jdbc-sessions/pom.xml [hunk 3]
```diff
@@ -46,19 +53,61 @@
...
+    <dependency>
+      <groupId>org.mariadb.jdbc</groupId>
+      <artifactId>mariadb-java-client</artifactId>
+      <version>2.6.0</version>
       <scope>test</scope>
     </dependency>
   </dependencies>
+  <profiles>
+    <profile>
+      <id>remote-session-tests</id>
```
Comment (janbartel) [inline tests/test-sessions/test-jdbc-sessions/pom.xml]: The only qualm I have here is that before, the jdbc session tests were being executed automatically but now we need this property set. Just a qualm.

**Review by janbartel — APPROVED** [review]
If you delete the last derby system property set, then I think we're good. Made a couple of comments, but they're just observations really.

File: tests/test-sessions/test-jdbc-sessions/pom.xml (hunk 3 above)
Comment (olamy) [reply tests/test-sessions/test-jdbc-sessions/pom.xml]: yup makes sense. I will only add an option to check if docker is available.

File: tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/SessionTableSchemaTest.java (hunk 2 above)
Comment (olamy) [reply tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/SessionTableSchemaTest.java:122]: ah yes I added that before truncate table after every tests.

**Pull request merged**